src/backend/partaj/core/models/referral.py

```diff
@@ -473,6 +475,29 @@ def unassign_unit(self, assignment, created_by):
         )
         return self.state
 
+    def change_urgencylevel(
```

Comment:
I think we can add state constraints here: only `ASSIGNED` or `RECEIVED` referrals should have their urgency changed.